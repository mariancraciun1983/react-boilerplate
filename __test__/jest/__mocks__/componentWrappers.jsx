import React  from 'react';
import clone from 'clone-deep';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';

import mockStore  from './store';


const { fullStore } = mockStore;

const genericWrapper = (children) => {
  const mockStore = configureStore();
  const initialState = clone(fullStore);
  const store = mockStore(initialState);
  return (
    <Provider store={store}>
      <Router>
        {children}
      </Router>
    </Provider>
  )
};

export {
  genericWrapper
};
