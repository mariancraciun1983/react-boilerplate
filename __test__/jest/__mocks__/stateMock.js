export default {
  app: {
    state: "loaded"
  },
  auth: {
    authenticated: true,
    token: "abc123",
    user: {
      name: "John Doe",
      email: "john@example.com",
      avatar: "https://dummyimage.com/100x100.png/5fa2d1/ffffff"
    }
  },
  genres: {
    list: [
      {
        name: "Comedy",
        slug: "comedy"
      },
      {
        name: "Adventure",
        slug: "adventure"
      }
    ]
  },
  movies: {
    list: [
      {
        id: "0b7b8b05-6c88-4517-a969-adf89a46bc0a",
        title: "Lost Palace",
        genre: ["comedy"],
        image: "http://dummyimage.com/100x100.png/cc0000/ffffff",
        description:
          "Donec diam neque, vestibulum eget, vulputate ut, ultrices vel, augue. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Donec pharetra, magna vestibulum aliquet ultrices, erat tortor sollicitudin mi, sit amet lobortis sapien sapien non mi. Integer ac neque.\n\nDuis bibendum. Morbi non quam nec dui luctus rutrum. Nulla tellus.\n\nIn sagittis dui vel nisl. Duis ac nibh. Fusce lacus purus, aliquet at, feugiat non, pretium quis, lectus.\n\nSuspendisse potenti. In eleifend quam a odio. In hac habitasse platea dictumst.\n\nMaecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat.",
        price: 2.39,
        slug: "gunner-palace",
        title_lower: "lost palace"
      },
      {
        id: "f01187b5-6fe8-465b-a0b8-3764a94ae225",
        title: "Love is Eternal While It Lasts",
        genre: ["comedy", "romance"],
        image: "http://dummyimage.com/100x100.png/5fa2dd/ffffff",
        description:
          "Curabitur gravida nisi at nibh. In hac habitasse platea dictumst. Aliquam augue quam, sollicitudin vitae, consectetuer eget, rutrum at, lorem.\n\nInteger tincidunt ante vel ipsum. Praesent blandit lacinia erat. Vestibulum sed magna at nunc commodo placerat.\n\nPraesent blandit. Nam nulla. Integer pede justo, lacinia eget, tincidunt eget, tempus vel, pede.",
        price: 9.5,
        slug: "love-is-eternal-while-it-lasts",
        title_lower: "love is eternal while it lasts"
      },
      {
        id: "23ad0474-8b8d-420d-a2cc-dd54aa6ed589",
        title: "Lost World, The",
        genre: ["romance"],
        image: "http://dummyimage.com/100x100.png/cc0000/ffffff",
        description:
          "In congue. Etiam justo. Etiam pretium iaculis justo.\n\nIn hac habitasse platea dictumst. Etiam faucibus cursus urna. Ut tellus.\n\nNulla ut erat id mauris vulputate elementum. Nullam varius. Nulla facilisi.\n\nCras non velit nec nisi vulputate nonummy. Maecenas tincidunt lacus at velit. Vivamus vel nulla eget eros elementum pellentesque.\n\nQuisque porta volutpat erat. Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla. Nunc purus.",
        price: 7.4,
        slug: "lost-world-the",
        title_lower: "lost world, the"
      }
    ]
  },
  cart: {
    dict: {
      "0b7b8b05-6c88-4517-a969-adf89a46bc0a": 1,
      "23ad0474-8b8d-420d-a2cc-dd54aa6ed589": 2
    },
    saved: false
  },
  filters: {
    string: "Lo",
    string_lower: "lo"
  },
  ui: {
    language: "en",
    version: 2,
    theme: "light"
  },
  routing: {
    locationBeforeTransitions: {
      pathname: "/genre/comedy",
      search: "",
      hash: "",
      key: "sb1up4"
    }
  },
  form: {}
};
