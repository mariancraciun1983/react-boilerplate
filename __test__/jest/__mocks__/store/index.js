import clone from 'clone-deep';

import * as FullData from '../../../../__mock__/api';

const defaultStore = {
  app: {
    state: 'loading',
  },
  auth: {
    authenticated: false,
    token: null,
    user: null,
  },
  cart: {
    dict: {},
    saved: true,
  },
  filters: {
    string: '',
    string_lower: '',
  },
  genres: {
    list: null,
  },
  movies: {
    list: null,
  },
  ui: {
    language: 'en',
    version: 0,
    theme: 'light',
  },
  routing: {},
  form: {},
};

const fullStore = clone(defaultStore);
fullStore.state = 'loaded';
fullStore.genres.list = FullData.genres;
fullStore.movies.list = FullData.movies.slice(0,200);
export default {
  defaultStore,
  fullStore,
}
