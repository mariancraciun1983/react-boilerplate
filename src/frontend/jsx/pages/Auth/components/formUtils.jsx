/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import className from 'classnames';

const getComponentToRender = (type, params = {}) => {
  if (type === 'textarea') {
    return <textarea {...params} />;
  }
  return <input {...params} type={type} />;
};

export const renderField = (dt) => {
  const {
    input,
    autoComplete,
    label,
    type,
    meta: { touched, error },
  } = dt;
  const hasError = touched && error;
  const hasSuccess = touched && !error;
  const params = {
    ...input,
    className: className({ 'form-control': true, 'is-valid': hasSuccess, 'is-invalid': hasError }),
    placeholder: label,
  };
  if (autoComplete) params.autoComplete = autoComplete;
  return (
    <div>
      <div className="form-group">
        {label ? <label htmlFor={params.name}>{label}</label> : null}
        {getComponentToRender(type, params)}
        {hasError ? <div className="invalid-feedback">{error}</div> : null}
      </div>
    </div>
  );
};

export const onSubmitFail = (errors) => {
  const key = Object.keys(errors)
    .reduce((k, l) => {
      if (document.getElementsByName(k)[0].offsetTop < document.getElementsByName(l)[0].offsetTop) {
        return k;
      }
      return l;
    });
  if (document.getEleentsByName(key).length) {
    document.getElementsByName(key)[0].focus();
  }
};
