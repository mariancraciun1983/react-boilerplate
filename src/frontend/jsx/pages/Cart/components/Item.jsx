/* eslint-disable radix */
import React from 'react';
import PropTypes from 'prop-types';
import { formatPrice } from '../../../../core/utils/formatter';

class Item extends React.PureComponent {
  constructor(props) {
    super(props);
    this.onUpdateMovie = this.onUpdateMovie.bind(this);
    this.onDelete = this.onDelete.bind(this);
  }

  onUpdateMovie(e) {
    const { onUpdate, movie: { id } } = this.props;
    onUpdate(id, parseInt(e.target.value));
  }

  onDelete() {
    const { onUpdate, movie: { id } } = this.props;
    onUpdate(id, 0);
  }

  render() {
    const { quantity, movie } = this.props;
    return (
      <div className="d-flex mb-3 pb-3 border-bottom">
        <div className="mr-auto d-inline-flex">
          <img src={movie.image} alt="Movie" />
          <div className="p-3">
            <strong>{movie.title}</strong>
            <br />
            <span>{`Price: ${formatPrice(movie.price)}$`}</span>
          </div>
        </div>
        <div className="">
          Quantity:
          <input
            className="form-control"
            type="number"
            value={quantity}
            min={1}
            onChange={this.onUpdateMovie}
          />
        </div>
        <div className="pt-4 pl-4">
          <button className="btn btn-dark" type="button" onClick={this.onDelete}>
            Delete
          </button>
        </div>
      </div>
    );
  }
}

Item.propTypes = {
  onUpdate: PropTypes.func.isRequired,
  movie: PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
  }).isRequired,
  quantity: PropTypes.number.isRequired,
};
export const Component = Item;
export default Item;
