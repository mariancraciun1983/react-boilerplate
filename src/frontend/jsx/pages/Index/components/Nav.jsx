import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import { genresList } from '../../../../core/selectors';

const stateToProps = (state) => ({
  genres: genresList(state),
});
class Nav extends React.PureComponent {
  render() {
    const { genres, genre } = this.props;
    return (
      <ul className="nav flex-column">
        <li className="nav-item">
          <div className="nav-link">
            <strong>Genres</strong>
          </div>
        </li>
        {genres.map((g) => {
          const active = (genre && genre.slug === g.slug);
          const classes = classnames({
            'nav-link': true,
            active,
          });
          return (
            <li className="nav-item" key={g.slug}>
              <Link className={classes} to={`/genre/${g.slug}`}>
                {g.name}
                { active ? '*' : ''}
              </Link>
            </li>
          );
        })}
      </ul>
    );
  }
}

Nav.propTypes = {
  genres: PropTypes.arrayOf(PropTypes.object).isRequired,
  genre: PropTypes.shape({
    slug: PropTypes.string,
  }),
};
Nav.defaultProps = {
  genre: null,
};

export const Component = Nav;
export default connect(
  stateToProps,
  null,
)(Nav);
