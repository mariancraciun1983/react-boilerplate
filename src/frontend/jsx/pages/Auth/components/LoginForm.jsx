import React from 'react';
import PropTypes from 'prop-types';
import {
  Field, reduxForm, getFormValues, submit,
} from 'redux-form';
import { connect } from 'react-redux';
import { onSubmitFail, renderField } from './formUtils';

const validate = (values) => {
  const errors = {};
  if (!values.email) {
    errors.email = 'Email is required';
  } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,30}$/i.test(values.email)) {
    errors.email = 'Invalid email address';
  }
  if (!values.password) {
    errors.password = 'Password is required';
  }
  return errors;
};

const formStateToProps = (state) => ({
  formValues: getFormValues('loginForm')(state),
});
const formActionsToProps = (dispatch) => ({
  attemptSubmit: () => dispatch(submit('loginForm')),
});
class LoginForm extends React.PureComponent {
  isSubmitDisabled() {
    const { pristine, submitting, initialValues } = this.props;
    if (submitting) return true;
    if (initialValues.email && initialValues.password) return false;
    if (pristine) return true;
    return false;
  }

  render() {
    const { handleSubmit, error, submitSucceeded } = this.props;
    return (
      <div>
        {error ? <div className="alert alert-danger">{error}</div> : null}
        {submitSucceeded ? (
          <div className="alert alert-success">Login successful</div>
        ) : null}
        <form onSubmit={handleSubmit}>
          <Field
            name="email"
            label="Email"
            component={renderField}
            type="email"
            autoComplete="email"
          />
          <Field
            name="password"
            label="Password"
            component={renderField}
            type="password"
            autoComplete="current-password"
          />
          <div className=" ">
            <button
              type="submit"
              disabled={this.isSubmitDisabled()}
              className="btn btn-success btn-lg"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    );
  }
}


LoginForm.propTypes = {
  pristine: PropTypes.bool.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  submitSucceeded: PropTypes.bool.isRequired,
  submitting: PropTypes.bool.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  error: PropTypes.any,
  initialValues: PropTypes.shape({
    email: PropTypes.string,
    password: PropTypes.string,
  }),
};
LoginForm.defaultProps = {
  initialValues: { email: '', password: '' },
  error: null,
};

const LoginWithReduxForm = reduxForm({
  form: 'loginForm',
  validate,
  onSubmitFail,
})(LoginForm);
export const Component = LoginForm;
export default connect(
  formStateToProps,
  formActionsToProps,
)(LoginWithReduxForm);
