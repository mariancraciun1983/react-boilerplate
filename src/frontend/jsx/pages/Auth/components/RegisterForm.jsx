import React from 'react';
import PropTypes from 'prop-types';
import {
  Field,
  reduxForm,
  getFormValues,
  submit,
} from 'redux-form';
import { connect } from 'react-redux';
import { onSubmitFail, renderField } from './formUtils';

const validate = (values) => {
  const errors = {};
  if (!values.name) {
    errors.name = 'Name is required';
  } else if (!values.email) {
    errors.email = 'Email is required';
  } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,30}$/i.test(values.email)) {
    errors.email = 'Invalid email address';
  }
  if (!values.password) {
    errors.password = 'Password is required';
  }
  return errors;
};


const formStateToProps = (state) => ({
  formValues: getFormValues('registerForm')(state),
});
const formActionsToProps = (dispatch) => ({
  attemptSubmit: () => dispatch(submit('registerForm')),
});
class RegisterForm extends React.PureComponent {
  isSubmitDisabled() {
    const { pristine, submitting, initialValues } = this.props;
    if (submitting) return true;
    if (initialValues.email && initialValues.password && initialValues.name) return false;
    if (pristine) return true;
    return false;
  }

  render() {
    const {
      handleSubmit,
      error,
      submitSucceeded,
    } = this.props;
    return (
      <div>
        {error ? <div className="alert alert-danger">{error}</div> : null}
        {submitSucceeded ? <div className="alert alert-success">Register successful</div> : null}
        <form onSubmit={handleSubmit}>
          <Field name="name" label="Name" component={renderField} type="name" autoComplete="name" />
          <Field
            name="email"
            label="Email"
            component={renderField}
            type="email"
            autoComplete="email"
          />
          <Field
            name="password"
            label="Password"
            component={renderField}
            type="password"
            autoComplete="current-password"
          />
          <div className=" ">
            <button
              type="submit"
              disabled={this.isSubmitDisabled()}
              className="btn btn-success btn-lg"
            >
              Register
            </button>
          </div>
        </form>
      </div>
    );
  }
}

RegisterForm.propTypes = {
  pristine: PropTypes.bool.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  submitSucceeded: PropTypes.bool.isRequired,
  submitting: PropTypes.bool.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  error: PropTypes.any,
  initialValues: PropTypes.shape({
    email: PropTypes.string,
    name: PropTypes.string,
    password: PropTypes.string,
  }),
};
RegisterForm.defaultProps = {
  initialValues: { email: '', password: '', name: '' },
  error: null,
};

export const Component = RegisterForm;
const RegisterWithReduxForm = reduxForm({
  form: 'registerForm',
  validate,
  onSubmitFail,
})(RegisterForm);
export default connect(
  formStateToProps,
  formActionsToProps,
)(RegisterWithReduxForm);
