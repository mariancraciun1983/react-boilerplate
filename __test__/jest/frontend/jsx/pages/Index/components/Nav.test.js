/// <reference types="jest" />
import React from 'react';
import renderer from 'react-test-renderer';

import {Component} from '../../../../../../../src/frontend/jsx/pages/Index/components/Nav';
import { genericWrapper } from '../../../../../__mocks__/componentWrappers';
import mock from '../../../../../__mocks__/store';

describe("components/pages/Index/components/Main", () => {
  let props;
  beforeEach(() => {
    props =  {
      genres: mock.fullStore.genres.list,
      genre: null,
    }
  });

  it("Renders Correctly", () => {
    expect.assertions(1);
    const tree = renderer
    .create(genericWrapper(<Component {...props}/>))
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders Correctly - Selected", () => {
    expect.assertions(1);
    const tree = renderer
    .create(genericWrapper(<Component {...props} genre={props.genres[1]} />))
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

});
