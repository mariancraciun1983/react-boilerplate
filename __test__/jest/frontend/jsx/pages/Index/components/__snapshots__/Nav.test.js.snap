// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/Index/components/Main Renders Correctly - Selected 1`] = `
<ul
  className="nav flex-column"
>
  <li
    className="nav-item"
  >
    <div
      className="nav-link"
    >
      <strong>
        Genres
      </strong>
    </div>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/action"
      onClick={[Function]}
    >
      Action
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link active"
      href="/genre/adventure"
      onClick={[Function]}
    >
      Adventure
      *
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/animation"
      onClick={[Function]}
    >
      Animation
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/children"
      onClick={[Function]}
    >
      Children
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/comedy"
      onClick={[Function]}
    >
      Comedy
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/crime"
      onClick={[Function]}
    >
      Crime
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/documentary"
      onClick={[Function]}
    >
      Documentary
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/drama"
      onClick={[Function]}
    >
      Drama
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/fantasy"
      onClick={[Function]}
    >
      Fantasy
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/horror"
      onClick={[Function]}
    >
      Horror
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/imax"
      onClick={[Function]}
    >
      IMAX
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/musical"
      onClick={[Function]}
    >
      Musical
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/mystery"
      onClick={[Function]}
    >
      Mystery
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/romance"
      onClick={[Function]}
    >
      Romance
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/scifi"
      onClick={[Function]}
    >
      Sci-Fi
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/thriller"
      onClick={[Function]}
    >
      Thriller
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/war"
      onClick={[Function]}
    >
      War
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/western"
      onClick={[Function]}
    >
      Western
      
    </a>
  </li>
</ul>
`;

exports[`components/pages/Index/components/Main Renders Correctly 1`] = `
<ul
  className="nav flex-column"
>
  <li
    className="nav-item"
  >
    <div
      className="nav-link"
    >
      <strong>
        Genres
      </strong>
    </div>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/action"
      onClick={[Function]}
    >
      Action
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/adventure"
      onClick={[Function]}
    >
      Adventure
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/animation"
      onClick={[Function]}
    >
      Animation
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/children"
      onClick={[Function]}
    >
      Children
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/comedy"
      onClick={[Function]}
    >
      Comedy
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/crime"
      onClick={[Function]}
    >
      Crime
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/documentary"
      onClick={[Function]}
    >
      Documentary
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/drama"
      onClick={[Function]}
    >
      Drama
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/fantasy"
      onClick={[Function]}
    >
      Fantasy
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/horror"
      onClick={[Function]}
    >
      Horror
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/imax"
      onClick={[Function]}
    >
      IMAX
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/musical"
      onClick={[Function]}
    >
      Musical
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/mystery"
      onClick={[Function]}
    >
      Mystery
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/romance"
      onClick={[Function]}
    >
      Romance
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/scifi"
      onClick={[Function]}
    >
      Sci-Fi
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/thriller"
      onClick={[Function]}
    >
      Thriller
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/war"
      onClick={[Function]}
    >
      War
      
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/genre/western"
      onClick={[Function]}
    >
      Western
      
    </a>
  </li>
</ul>
`;
